docs/vectors/sealedbox_vectors.rst

```diff
@@ -0,0 +1,10 @@
+SealedBox reference vectors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Since, at the time of writing, we haven't been able to find
```

Comment:
@maqp I really like the way you've rewritten the section; thank you.

As for your the stable reference vector generation, I'm unsure if we should really rely on long-term stability of libsodium's `randombytes_buf_deterministic`, and I'd rather have the users just test correct round-trip behaviour.

As for avoiding the need to compile and link the executable, I think the need to generate and test vectors is advanced enough to let us require the user to be able to compile and link the driver by herself.